### PR TITLE
Feature/as a user i should be able to sign in to the system using my email password

### PR DESCRIPTION
**Pull request by frankvielma**
The PR adds a Sign In to the system using my email & password based on the criteria in this [card](https://trello.com/c/gTAGip8R/10-as-a-user-i-should-be-able-to-sign-in-to-the-system-using-my-email-password).

**Acceptance Criteria**

- [x] Given that I am signed out
- [x] When I go to the homepage
---
- [x] Then I should land on the sign in page and see a form to sign in
- [x] When I fill the form correctly
- [x]  Then I should be signed in and be redirected to the signed in homepage
- [x] When I fill the form incorrectly
- [x]  Then I should see an error message indicating my mistakes
---
- [x] Then I should Also see a sign up link in case I don't have an account
- [x] When I click it
- [x] Then I should be redirected to the sign up page

[video_signup.webm](https://github.com/frankvielma/rs-blackmarket-fv/assets/412081/d5f20365-e70b-41f7-83df-0b9de2c10eab)

### CODE REVIEW

File: app/login/page.tsx [hunk 1]
```diff
@@ -33,11 +39,39 @@ export default function Login() {
...
+      },
+      body: JSON.stringify(data),
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        if (data['access_token'] !== undefined) {
+          router.push('/dashboard')
+        } else {
+          if (
+            data['non_field_errors'] ==
```
Comment (dariokozicki) [inline app/login/page.tsx:66]: Whenever possible, use immutability (don't change the values of responses)

Comment (dariokozicki) [reply app/login/page.tsx:66]: Also comparing with error descriptions in code has the risk of stopping working if someone changes the error message, in these cases (if possible) use status codes

**Review by dariokozicki — APPROVED** [review]
Great job!

File: app/login/page.tsx (hunk 1 above)
Comment (frankvielma) [reply app/login/page.tsx:66]: :+1:

**Pull request merged**